frontend/src/utils/fieldValidation.ts

```diff
@@ -285,6 +296,62 @@ export const createCheckboxValidationRules: ValidationRuleFn<
   }
 }
 
+export const createDateValidationRules: ValidationRuleFn<DateFieldBase> = (
+  schema,
+) => {
+  return {
+    ...createBaseValidationRules(schema),
+    validate: {
+      validDate: (val) =>
+        !val || isDate(parseISO(val)) || 'Please enter a valid date',
+      noFuture: (val) => {
+        if (
+          !val ||
+          schema.dateValidation.selectedDateValidation !==
+            DateSelectedValidation.NoFuture
+        ) {
+          return true
+        }
+        return (
+          isBefore(parseISO(val), startOfTomorrow()) ||
+          'Only dates today or before are allowed'
+        )
+      },
+      noPast: (val) => {
+        if (
+          !val ||
+          schema.dateValidation.selectedDateValidation !==
+            DateSelectedValidation.NoPast
+        ) {
+          return true
+        }
+        return (
+          isAfter(parseISO(val), endOfYesterday()) ||
```

Comment:
should we also make this consistent with `isDateUnavailable`?

Reply:
perhaps we can extract this into a helper so future devs don't have to keep wrapping their head around "is `isAfter(parseISO(val), endOfYesterday())` equivalent to `!isBefore(date, startOfToday())`"